### ISSUE TITLE
macOS: all builds failing in all repos.

### ISSUE THREAD

**Issue by evoskuil** [libbitcoin/libbitcoin-build#317]
https://github.com/libbitcoin/libbitcoin-system/actions/runs/8916066422

**Comment by evoskuil** [comment]
```
/Users/runner/work/libbitcoin-system/prefix/include/boost/container_hash/hash.hpp:132:33: error: no template named 'unary_function' in namespace 'std'; did you mean '__unary_function'?
        struct hash_base : std::unary_function<T, std::size_t> {};
                           ~~~~~^
/Applications/Xcode_15.0.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/usr/include/c++/v1/__functional/unary_function.h:46:1: note: '__unary_function' declared here
using __unary_function = __unary_function_keep_layout_base<_Arg, _Result>;
^
1 error generated.
```

https://github.com/libbitcoin/libbitcoin-system/actions/runs/8916066422/job/24486829288

**Comment by pmienk** [comment]
Was resolved by #322
